sdk/mixedreality/azure-mixedreality-authentication/azure/mixedreality/authentication/aio/_client_async.py

```diff
@@ -0,0 +1,112 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+# pylint: disable=unused-import,ungrouped-imports
+from typing import Any, TYPE_CHECKING, Union
+
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse # type: ignore
+
+from azure.core.credentials import AzureKeyCredential
+from azure.core.tracing.decorator_async import distributed_trace_async
+from azure.core.pipeline.policies import AsyncBearerTokenCredentialPolicy
+
+from .._generated.aio import MixedRealityStsRestClient
+from .._generated.models import TokenRequestOptions
+from .._version import SDK_MONIKER
+from .._shared.authentication_endpoint import construct_endpoint_url
+from .._shared.aio.mixedreality_account_key_credential import MixedRealityAccountKeyCredential
+from .._utils import convert_to_access_token, generate_cv_base
+
+if TYPE_CHECKING:
+    from azure.core.credentials import AccessToken
+    from azure.core.credentials_async import AsyncTokenCredential
+
+
+class MixedRealityStsClient(object):
+    """ A client to interact with the Mixed Reality STS service.
+
+    :param str account_id:
+        The Mixed Reality service account identifier.
+    :param str account_domain:
+        The Mixed Reality service account domain.
+    :param Union[TokenCredential, AzureKeyCredential] credential:
+        The credential used to access the Mixed Reality service.
+    :keyword str custom_endpoint_url:
+        Override the Mixed Reality STS service endpoint.
+    """
+
+    def __init__(self,
+        account_id: str,
+        account_domain: str,
+        credential: Union[AzureKeyCredential, "AsyncTokenCredential"], #pylint: disable=unsubscriptable-object
+        **kwargs) -> None:
+        if not account_id:
+            raise ValueError("account_id can not be None")
+
+        if not account_domain:
+            raise ValueError("account_domain can not be None")
+
+        if not credential:
+            raise ValueError("credential can not be None")
+
+        self._account_id = account_id
+        self._account_domain = account_domain
+
+        if isinstance(credential, AzureKeyCredential):
+            credential = MixedRealityAccountKeyCredential(account_id, credential)
+
+        self._credential = credential
+
+        endpoint_url = kwargs.pop('custom_endpoint_url', construct_endpoint_url(account_domain))
```

Comment:
If the `custom_endpoint_url` is used in place of `account_domain` - Why do I still need to provide an `account_domain`? What else would that be used for apart from constructing the URL?

Reply:
Today, it's not used for anything else. I wanted to keep it to keep things simple and consistent with the other language SDKs. I'm open to alternatives if there's a more Python'y way of doing it. We expect 95% of customers to use the `accountDomain` and to never have to think about the endpoint url.

Reply:
What about national clouds?

Reply:
I don't know what "national clouds" are.

Reply:
Ah sorry - I mean the national clouds in China, Germany, Government clouds etc.
Is the service supported there? And if so, do authenticate using the same `account_domain` parameter?

Reply:
I don't believe so, but either way, if there was something custom there were we couldn't properly derive the endpoint from the account domain, they would need to specify the custom_endpoint_url. But, that would likely only ever be done with handholding from us.

Reply:
As discussed offline, we don't expect many people to use this library directly and even if they do, the custom endpoint should satisfy the edge cases.